packages/workspace/src/utils/graph-utils.ts

```diff
@@ -17,44 +17,49 @@ const reach: Reach = {
 };
 
 function buildMatrix(graph: ProjectGraph) {
-  const dependencies = graph.dependencies;
   const nodes = Object.keys(graph.nodes);
+  const nodesLength = nodes.length;
   const adjList = {};
   const matrix = {};
 
-  const initMatrixValues = nodes.reduce((acc, value) => {
-    return {
-      ...acc,
-      [value]: false,
-    };
-  }, {});
-
-  nodes.forEach((v) => {
+  // create matrix value set
+  for (let i = 0; i < nodesLength; i++) {
+    const v = nodes[i];
     adjList[v] = [];
-    matrix[v] = { ...initMatrixValues };
-  });
+    // meeroslav: turns out this is 10x faster than spreading the pre-generated initMatrixValues
+    const nodeValues = {};
+    for (let j = 0; j < nodesLength; j++) {
+      nodeValues[nodes[j]] = false;
```

Comment:
We don't need to initiate the nodeValues with false, but could leave them simply undefined. On my big repo with almost 3000 projects, not initiate the nodeValue saves 1300ms.

The code would then look like:
`
for (let i = 0; i < nodesLength; i++) {
        const v = nodes[i];
        adjList[v] = [];
        matrix[v] = {};
    }
`
We need to make an adjustment for the traverse function as well. This could be written as:
`
const traverse = (s: string, v: string) => {
      matrix[s][v] = true;
      const adjListLength = adjList[v].length;
      for (let i = 0; i < adjListLength; i++) {
        const adj = adjList[v][i];
        if (!matrix[s][adj]) {
          traverse(s, adj);
        }
      }
    };
`